src/main/java/lotto/domain/LottoResult.java

```diff
@@ -0,0 +1,36 @@
+package lotto.domain;
+
+import java.util.Arrays;
+
+public enum LottoResult {
+
+    FIRST(LottoNumberMatchCount.ALL, Money.wons(2_000_000_000)),
+    SECOND(LottoNumberMatchCount.FIVE, Money.wons(1_500_000)),
+    THIRD(LottoNumberMatchCount.FOUR, Money.wons(50_000)),
+    FOURTH(LottoNumberMatchCount.THREE, Money.wons(5_000)),
+    NOTHING(LottoNumberMatchCount.NOTHING, Money.ZERO);
+
+    private final LottoNumberMatchCount lottoNumberMatchCount;
+    private final Money prizeMoney;
+
+    LottoResult(final LottoNumberMatchCount lottoNumberMatchCount, final Money prizeMoney) {
+        this.lottoNumberMatchCount = lottoNumberMatchCount;
+        this.prizeMoney = prizeMoney;
+    }
+
+    public static LottoResult fromLottoNumberMatchCount(
+        final LottoNumberMatchCount lottoNumberMatchCount) {
+        return Arrays.stream(values())
+            .filter(result -> result.lottoNumberMatchCount.equals(lottoNumberMatchCount))
```

Comment:
```suggestion
            .filter(result -> result.isMatchCount(lottoNumberMatchCount))
```

데이터를 물어보기 보다 직접 비교하기 보다는 객체에게 비교하도록 시켜본다면 어떨까요?

[[객체지향 생활체조 원칙] 규칙 9. getter/setter/property를 쓰지 않는다](https://limdingdong.tistory.com/15)

Reply:
세세한 부분까지 캐치 감사합니다 :) 다음 스텝에서 반영하도록 할께요

Reply:
찾아보니 해당 클래스 내부 필드라서 프로퍼티 접근을 한 것 같습니다만
클래스 내부 프로퍼티를 이용할때도 getter쓰는게 맞을까용..? 
좋은 의견 있으시면 공유 부탁 드립니다!